### PR TITLE
DApp-1543 blocks/box component 

### PR DESCRIPTION
**Pull request by rohitmalhotra1420**
### Ticket Number

- [1543](https://github.com/push-protocol/push-dapp/issues/1543)

### Description

- **Problem/Feature**:
- I have introduced a layout component from the name of Box.
- It is loosely based on Chakra UI and Radix UI Box component.
- The idea here is to standardise our codebase base as well as development process, to follow similar approach for UI development. To avoid the usage of style props directly and use styled component classes. So we can have better code readability and avoid creating multiple styled components.
- I have added couple of css props to this component, but new features and fixes will be rolled out of it soon as we move ahead. 
- This will not support style prop to avoid inline styling.
- In case there is a sheer need to apply additional css which is not supported by Box or we don't want to add it's support to Box then we can use styled components `css` prop to pass down extra css.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

- [ ] **Quick PR**: Is this a quick PR? Can be approved before finishing a coffee.
  - [ ] Quick PR label added
- [ ] **Not Merge Ready**: Is this PR dependent on some other PR/tasks and not ready to be merged right now.
  - [ ] DO NOT Merge PR label added

### Frontend Guidelines

- [x] Followed frontend guidelines as per the [Guidelines Notion Doc](https://www.notion.so/pushprotocol/Frontend-dApp-Guidelines-1d7806ae3d9e4569a340b563dcd0536c)

### Build & Testing

- [x] No errors in the build terminal
- [x] Engineer has tested the changes on their local environment
- [x] Engineer has tested the changes on deploy preview

### Screenshots/Video with Explanation

- **Before:** Explain the previous behavior

- **After:** What's changed now

### Review & Approvals

- [x] Self-review completed
- [ ] Code review by at least one other engineer
- [ ] Documentation updates if applicable

### CODE REVIEW

File: src/blocks/box/Box.types.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  height?: ResponsiveProp<string>;
+  justifyContent?: ResponsiveProp<CSSProperties['justifyContent']>;
+  m?: ResponsiveProp<SpaceType>;
+  maxH?: ResponsiveProp<string>;
+  minH?: ResponsiveProp<string>;
+  maxW?: ResponsiveProp<string>;
+  minW?: ResponsiveProp<string>;
+  p?: ResponsiveProp<SpaceType>;
+  width?: ResponsiveProp<string>;
+};
```
Comment (abhishek-01k) [inline src/blocks/box/Box.types.ts:18]: - flex direction should also be added here as reponsive prop.

**Comment by abhishek-01k** [comment]
Rest everything looks good and working

File: src/blocks/box/Box.types.ts (hunk 1 above)
Comment (rohitmalhotra1420) [reply src/blocks/box/Box.types.ts:18]: Will add it.

Comment (mishramonalisha76) [reply src/blocks/box/Box.types.ts:18]: There should be more properties added like background and border, but we can add them as needed i guess

Comment (rohitmalhotra1420) [reply src/blocks/box/Box.types.ts:18]: Both are already there

Comment (rohitmalhotra1420) [reply src/blocks/box/Box.types.ts:18]: https://github.com/push-protocol/push-dapp/pull/1590/files#diff-7d949045d17e37e448062b4981686c38b96e2c3f5177aaae4838755ca622b7bcR20-R28

**Review by mishramonalisha76 — DISMISSED** [review]

**Review by abhishek-01k — APPROVED** [review]

**Review by abhishek-01k — APPROVED** [review]

**Review by mishramonalisha76 — APPROVED** [review]

**Pull request merged**